app/controllers/SessionTimeoutPageController.scala

```diff
@@ -0,0 +1,141 @@
+import models.ui.Text
+import services.GuidanceService
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import uk.gov.hmrc.http.logging.SessionId
+import uk.gov.hmrc.http.SessionKeys._
+import views.html.session_timeout
+
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+@Singleton
+class SessionTimeoutPageController @Inject()(
+                                              appConfig: AppConfig,
+                                              service: GuidanceService,
+                                              errorHandler: ErrorHandler,
+                                              mcc: MessagesControllerComponents,
+                                              view: session_timeout)
+    extends FrontendController(mcc)
+    with I18nSupport {
+
+    val logger = Logger(getClass)
+
+    def getPage(processCode: String): Action[AnyContent] = Action.async { implicit request =>
+
+      implicit val messages: Messages = mcc.messagesApi.preferred(request)
+
+      val sessionId: Option[SessionId] = hc.sessionId
+
+      sessionId match {
+        case Some(id) => {
+          if(hasSessionExpired(request.session)) {
+            Future.successful(createYourSessionHasExpiredResponse(messages("session.timeout.header.title"), processCode))
+          } else {
+            manageDeleteYourAnswersRequest(id, processCode)
+          }
+        }
+        case None => Future.successful(createYourSessionHasExpiredResponse(messages("session.timeout.header.title"), processCode))
+      }
+    }
+
+  def manageDeleteYourAnswersRequest(sessionId: SessionId, processCode: String)(implicit request: Request[_]): Future[Result] = {
+
+    // Refresh cache to remove answers
+    implicit val messages: Messages = mcc.messagesApi.preferred(request)
+
+    service.getProcessContext(sessionId.value).flatMap {
+      case Right(processContext) if (processCode != processContext.process.meta.processCode) => {
+        logger.error(s"Unexpected process code encountered when removing session after timeout warning. Expected code $processCode; actual code ${processContext.process.meta.processCode}")
+        Future.successful(InternalServerError(errorHandler.internalServerErrorTemplate))
+      }
+      case Right(processContext) => {
+        service.removeSession(sessionId.value).map {
+            case Left(err) => {
+              logger.error(s"Error $err occurred attempting to remove session ${sessionId.toString}")
+              InternalServerError(errorHandler.internalServerErrorTemplateWithProcessCode(Some(processCode)))
+            }
+            case Right(_) => {
+              val title = Text(processContext.process.title.langs)
+              createDeleteYourAnswersResponse(title.asString(messages.lang), processCode)
+            }
+        }
+      }
+      case Left(NotFoundError) => {
+        // For some reason the timeout mechanism invokes the "getPage" method twice. In this case
```

Comment:
I think with the conditional inclusion of the timeout dialog, this code is no longer required.

Reply:
removed